main.js

```diff
@@ -14,4 +20,208 @@ mobileMenuOptions.forEach((option) => {
+    articleP.appendChild(lang);
+
+    for (let j = 0; j < myWorksArray[i].technologies.length; j += 1) {
+      const language = document.createElement('li');
+      language.className += 'language';
+      language.innerHTML = myWorksArray[i].technologies[j];
+      lang.appendChild(language);
+    }
+
+    const butt = document.createElement('div');
+    butt.className += 'butt';
+    articleP.appendChild(butt);
+
+    const btn = document.createElement('button');
+    btn.className += 'btn';
+    btn.innerHTML = ' See Project';
+    butt.appendChild(btn);
+
+    btn.addEventListener('click', () => {
+      popupSection.classList.remove('hide');
+      const contains = `<div class="card">
+      <div class="card-header">
+        <img class="close-button" id="closeButton" onClick="closePopup();" src="images/icons/cloasepopup.svg" alt="close Button"/>
+      </div>
+      <div class="card-body">
+        <img src="${
+  myWorksArray[i].featuredImage
+}" alt="my work Portrait"/>
+        <div class="two-columns">
+          <h4>Keeping track of hundreds of
+            components</h4>
+          <div class="display-centered">
+            <a href="" class="btn-secondary">See Live <img src="images/icons/live.svg" alt="See PRoject Live"/></a>
+            <a href="" class="btn-secondary">See Source <img src="images/icons/github.svg" alt="See Source Code"/></a>
+          </div>
+        </div>
+        <div class="childs">
+          <div class="asscories">
+            <ul class="card-tags">`;
+
+      let newContains = '';
+      for (let j = 0; j < myWorksArray[i].technologies.length; j += 1) {
+        const licontains = `<li>${myWorksArray[i].technologies[j]}</li>`;
+        newContains += licontains;
+      }
+      const conatins2 = `</ul>
+             
+          </div>
+          <p>${
+  myWorksArray[i].description
+}</p>
+          <div class="mobile-buttons">
+            <a href="${
+  myWorksArray[i].linkToLiveVersion
+}" class="btn-secondary">See Live <img src="images/icons/live.svg" alt="See PRoject Live"/></a>
+            <a href="${
+  myWorksArray[i].linkToSource
+}" class="btn-secondary">See Source <img src="images/icons/github.svg" alt="See Source Code"/></a>
+          </div>
+        </div>
+      </div>
+    </div>`;
+
+      popupSection.innerHTML += contains + newContains + conatins2;
+      const closeButton = document.getElementById('closeButton');
+      closeButton.addEventListener('click', () => {
+        closePopup();
+      });
+    });
+  }
+});
+
+const getData = JSON.parse(localStorage.getItem('formData'));
+if (getData != null) {
+  nameElement.value = getData.name;
+  emailElement.value = getData.email;
+  msgElement.value = getData.message;
+}
+
+// Save Data TO Local Storage
+function saveToLocalStorage() {
+  const formData = {
+    name: nameElement.value,
+    email: emailElement.value,
+    message: msgElement.value,
+  };
+  localStorage.setItem('formData', JSON.stringify(formData));
+}
```

Comment:
I can see you have done a great job setting up the local storage functionality😉😉but there are still some issues according to the requirements of this milestone
- [ ] It is a[ requirement](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md) that when the user changes the content of any input field and when the user reloads the page, the data is saved to the local storage, currently, the local storage is not populated with the form data when the content is changed. Kindly fix that😉
- [ ] Also, the local storage isn't been populated with the input data😉 kindly fix that as it is a core [requirement](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md) for this milestone🤗🤗

Reply:
done